From 66bec2dd8865f139be0555994c122a8e414fd353 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:01:36 -0400
Subject: [fs] btrfs: cancel scrub on transaction abortion

Message-id: <1407179109-13714-210-git-send-email-sandeen@redhat.com>
Patchwork-id: 87128
O-Subject: [RHEL7.1 PATCH 209/422] Btrfs: cancel scrub on transaction abortion
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Wang Shilong <wangsl.fnst@cn.fujitsu.com>

commit c0af8f0b1cf7ec5cde4450be9f8bfeb8c211d40a
Author: Wang Shilong <wangsl.fnst@cn.fujitsu.com>
Date:   Wed Feb 19 19:24:18 2014 +0800

    Btrfs: cancel scrub on transaction abortion

    If we fail to commit transaction, we'd better
    cancel scrub operations.

    Suggested-by: Miao Xie <miaox@cn.fujitsu.com>
    Signed-off-by: Wang Shilong <wangsl.fnst@cn.fujitsu.com>
    Signed-off-by: Josef Bacik <jbacik@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/transaction.c b/fs/btrfs/transaction.c
index 84da666..79a4186 100644
--- a/fs/btrfs/transaction.c
+++ b/fs/btrfs/transaction.c
@@ -1580,6 +1580,7 @@ static void cleanup_transaction(struct btrfs_trans_handle *trans,
 
  if (current->journal_info == trans)
   current->journal_info = NULL;
+ btrfs_scrub_cancel(root->fs_info);
 
  kmem_cache_free(btrfs_trans_handle_cachep, trans);
 }
-- 
1.7.1